test/system/src/main/java/io/pravega/test/system/framework/Utils.java

```diff
@@ -149,18 +157,27 @@ public static Service createPravegaSegmentStoreService(final URI zkUri, final UR
     }
 
     public static ClientConfig buildClientConfig(URI controllerUri) {
-        if (!AUTH_ENABLED) {
-            log.debug("Generating config with auth disabled.");
-            return ClientConfig.builder().controllerURI(controllerUri).build();
-        } else {
+        if (TLS_AND_AUTH_ENABLED) {
```

Comment:
@ravisharda, is it possible to have only TLS enabled why AUTH is disabled? It looks a bit odd that both must be enabled or at least TLS cannot be enabled alone. I was told that there is something in the current design that forces so, could you please clarify a bit? Thanks

Reply:
@medvedevigorek technically both can be enabled/disabled independently of each other. So, you could have a config where TLS is enabled, but auth is disabled. However, in practical use, both should be enabled. 

The older flag in the system tests that represented "security enabled" only turned on `auth`, but not `TLS`, due to some of the inherent gaps that we had with the system tests configuration. With @ravibeta's changes, we can not have it turn on both `auth` and `tls`. 

With that background, as well as what the specific code in question does, it looked to appropriate to call it `TLS_AND_AUTH_ENABLED` as opposed to just `TLS_ENABLED`.

Hope that clarifies.

Reply:
okay, getting even more confused now. 

I get the part that in most cases `auth`  and `tls` are enabled together as it doesn't really make sense to have `auth` without `tls`. However there could be valid cases where `auth` is not required by `tls` is required. 

If in the current state we say that we always want to have both enabled then we have a bit confusing configuration: 
`securityEnabled` - enables only `auth` part
'tlsEnabled` - enables `tls` and `auth` (**this is what confuses me**) 

We need either
(a) `securityEnabled` - enables both `auth` and `tls`. And don't introduce `tlsEnabled` property at all 
(b) Have both properties present and require both to be enabled or none. But `tlsEnabled` shouldn't change `auth` state. 

What do you think?

Reply:
Yes Igor, it is too big a change for these system tests. May I please address it separately ? The issue is that pravega-operator deploys pravega cluster with the admin username and password and this can be fixed outside pravega repository first and accomodated in SDP. Thanks,